docs/source/geo_def.rst

```diff
@@ -88,10 +89,83 @@ based on area extent and a proj4-string or a list of proj4 arguments.
  Number of rows: 425
  Area extent: (-5326849.0625, -5326849.0625, 5326849.0625, 5326849.0625)
 
+The function **from_params** attempts to find shape and area_extent with the given data below:
+
+Required (positional) arguments:
+
+* **description**: Description
+* **projection**: projection parameters as a proj4_dict or proj4_string
+
+Optional (keyword) arguments:
 
-The **load_area** function can be used to parse area definitions from a
-configuration file. Assuming the file **areas.yaml** exists with the following
-content
+* **area_id**, **proj_id**, and **area_extent**: same as AreaDefinition
+* **units**: Default projection units (meters/radians/degrees)
+* **shape**: (height, width). Note: if height = width, then only height or width needs to be passed.
+* **top_left_extent**: (x_ul, y_ul)
+* **center**: (center_x, center_y)
+* **pixel_size**: (x_size, y_size). Note: if x_size = y_size, then only x_size or y_size needs to be passed.
+* **radius**: (x_length, y_length). Note: if x_length = y_length, then only x_length or y_length needs to be passed.
+
+where
+
+* **x_ul**: projection x coordinate of upper left corner of upper left pixel
+* **y_ul**: projection y coordinate of upper left corner of upper left pixel
+* **center_x**: projection x coordinate of center of projection
+* **center_y**: projection y coordinate of center of projection
+* **height**: number of pixels in y direction (number of grid rows)
+* **width**: number of pixels in x direction (number of grid columns)
+* **x_size**: projection size of pixels in the x direction
+* **y_size**: projection size of pixels in the y direction
+* **x_length**: projection length from center to left/right outer edge
+* **y_length**: projection length from center to top/bottom outer edge
+* **units** accepts anything with 'm', 'rad', 'deg' or '°'. The order of default is: units expressed
+  with each variable, units passed to **units**, units used in **projection**, then meters. **shape** is not affected
+  by units. To add units to a specific variable, you can make an xarray.DataArray with a variable and then use
+  {'units': unit_of_choice} as its attrs attribute.
```

Comment:
I'm not a huge fan of discussing this here in this document. It complicates things a lot. Might be better just to keep that information in the docstring.